dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -371,10 +382,9 @@ def runValidation(self, jobId, interfaces):
                             continue
 
                     if not passedValidations:
-                        if failures:
+                        if self.writeErrors(failures, interfaces, jobId, shortColnames, writer, warningWriter, rowNumber):
                             rowErrorPresent = True
                             errorRows.append(rowNumber)
```

Comment:
Good catch, my intent was that warnings would not make a row invalid, since they will still be able to submit those rows.  I've added a check in the runSqlValidations function to only include rows with a fatal error in the returned list.